src/libs/ajax/leonardo/providers/LeoAppProvider.test.ts

```diff
@@ -0,0 +1,97 @@
+import { Ajax } from 'src/libs/ajax';
+import { AppAjaxContract, AppsAjaxContract } from 'src/libs/ajax/leonardo/Apps';
+import { asMockedFn } from 'src/testing/test-utils';
+
+import { leoAppProvider } from './LeoAppProvider';
+
+jest.mock('src/libs/ajax');
+
+type AjaxContract = ReturnType<typeof Ajax>;
+type AppNeeds = Pick<AppAjaxContract, 'delete' | 'pause'>;
+type AppsNeeds = Pick<AppsAjaxContract, 'app' | 'listWithoutProject'>;
+
+interface AjaxMockNeeds {
+  Apps: AppsNeeds;
+  app: AppNeeds;
+}
+/**
+ * local test utility - mocks the Ajax super-object and the subset of needed multi-contracts it
+ * returns with as much type-saftely as possible.
+ *
+ * @return collection of key contract sub-objects for easy
+ * mock overrides and/or method spying/assertions
+ */
+const mockAjaxNeeds = (): AjaxMockNeeds => {
+  const partialApp: AppNeeds = {
+    delete: jest.fn(),
+    pause: jest.fn(),
+  };
+  const mockApp = partialApp as AppAjaxContract;
+
+  const partialApps: AppsNeeds = {
+    app: jest.fn(),
+    listWithoutProject: jest.fn(),
+  };
+  const mockApps = partialApps as AppsAjaxContract;
+
+  asMockedFn(mockApps.app).mockReturnValue(mockApp);
+  asMockedFn(Ajax).mockReturnValue({ Apps: mockApps } as AjaxContract);
+
+  return {
+    Apps: partialApps,
+    app: partialApp,
+  };
+};
+describe('leoAppProvider', () => {
+  it('handles list call', async () => {
+    // Arrange
+    const ajaxMock = mockAjaxNeeds();
+    asMockedFn(ajaxMock.Apps.listWithoutProject).mockResolvedValue([]);
+    const abort = new window.AbortController();
```

Comment:
Style nit: I'd drop the "window" to make things a little more concise.

Reply:
really?  I've run into unit test harness issues if not being explicit about globals coming from window.whatever.  I've also read that it's generally a best practice.

Reply:
I think it's unnecessary verbosity.

What issues have you run into?

Other tests work ok without `window`.
https://github.com/DataBiosphere/terra-ui/blob/c8858c5063c796380b6dc18f48f42b8411001531/src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts#L566

And Terra UI itself sometimes accesses globals without using `window`.
https://github.com/DataBiosphere/terra-ui/blob/c8858c5063c796380b6dc18f48f42b8411001531/src/libs/ajax/incremental-response/useIncrementalResponse.ts#L19

Reply:
I've run into Typescript issues in the past, and also for edge-cases where the jest test environment lacks a specific global, having window prefix makes it easier to chase down what's wrong.

Reply:
I've seen the TypeScript issues when accessing globals that Terra UI defines, such as `ajaxOverrideUtils`. But the browser built-ins should be fine.

Reply:
that is a good distinction to make, but I still like explicitly prefixing with window. since it aids in adding context as I noted.